src/main/java/bookmarked/command/FindCommand.java

```diff
@@ -32,21 +37,37 @@ public FindCommand(String newItem, ArrayList<Book> listOfBooks) {
 
     @Override
     public void handleCommand() {
-        assert listOfBooks != null : "list of books should not be empty";
-        String keyword;
-        logger.log(Level.INFO, "going to start processing find command");
-        keyword = getKeyword();
-        if (keyword == null) {
-            return;
-        }
-
-        try {
-            processFind(keyword);
-        } catch (EmptyListException e) {
-            Ui.printEmptyListMessage();
+        this.splitCommand = newItem.split("/by");
+        String splitCommandCommand = splitCommand[1].trim();
+        String[] findItem = splitCommandCommand.split(" ");
+        switch (findItem[0].trim()) {
+            case ("book"):
+                assert listOfBooks != null : "list of books should not be empty";
+                String keyword;
+                logger.log(Level.INFO, "going to start processing find command");
+                keyword = getKeyword();
+                if (keyword == null) {
+                    return;
+                }
+                try {
+                    processFind(keyword);
+                } catch (EmptyListException e) {
+                    Ui.printEmptyListMessage();
+                }
+                break;
+            case ("user"):
+                String[] userName = splitCommandCommand.split("user");
+                if (userName.length >1) {
+                    FindUserCommand findUserCommand = new FindUserCommand(listOfUsers, userName[1].trim());
+                    findUserCommand.handleCommand();
+                } else {
+                    System.out.println("user name cannot be empty");
+                }
+                break;
         }
```

Comment:
what happen if argument after "/by" is neither book nor user?
example: `find /by`, `find /by  `, or `find /by nothing`